packages/grid/x-grid/src/tests/filtering.XGrid.test.tsx

```diff
@@ -198,6 +198,39 @@ describe('<XGrid /> - Filter', () => {
     expect(getColumnValues()).to.deep.equal(['Adidas', 'Puma']);
   });
 
+  it('should trigger onFilterModelChange when the link operator changes', () => {
+    const onFilterModelChange = spy();
+    const newModel: GridFilterModel = {
+      items: [
+        {
+          columnField: 'brand',
+          value: 'a',
+          operatorValue: 'startsWith',
+        },
+        {
+          columnField: 'brand',
+          value: 'a',
+          operatorValue: 'endsWith',
+        },
+      ],
+    };
+    render(
+      <TestCase
+        filterModel={newModel}
+        onFilterModelChange={onFilterModelChange}
+        state={{
+          preferencePanel: { openedPanelValue: GridPreferencePanelsValue.filters, open: true },
+        }}
+      />,
+    );
+    expect(onFilterModelChange.callCount).to.equal(4);
```

Comment:
Off-topic: should we batch all changes in the state during mount and only call `onFilterModelChange` once? It seems counterintuitive to dispatch a new event for each filter in the model.

Reply:
Yeah, and even more than colling it once. The value never change, so it should never fire:

```suggestion
    expect(onFilterModelChange.callCount).to.equal(0);
```

Reply:
@oliviertassinari is there a Ticket for this issue? `onFilterModelChange` is firing way to often and sometimes with an invalid default state. This makes it hard to control the `filterModel`.

Reply:
I worked on this in https://github.com/mui-org/material-ui-x/pull/1760/files#r642102787

Reply:
I have created an issue regarding this topic. https://github.com/mui-org/material-ui-x/issues/1985
This might help reproducing the issue in more detail. Thanks for your quick response 🙏!